backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/LlvmLinkOptions.kt

```diff
@@ -0,0 +1,11 @@
+package org.jetbrains.kotlin.backend.konan.llvm
+
+import llvm.LLVMAddNamedMetadataOperand
+import llvm.LLVMModuleRef
+
+fun embedLlvmLinkOptions(module: LLVMModuleRef, options: List<List<String>>) {
+    options.forEach {
```

Comment:
Shalln't we assert that it only works on Apple targets?

Reply:
This is kinda not true.

Reply:
You mean, we could use it on other platforms? Like to fix https://github.com/JetBrains/kotlin-native/issues/2803

Reply:
Maybe.
E.g. https://github.com/llvm-mirror/llvm/commit/e7676fec11b02e4b698b5ffc99e1901246a7bf66
Not sure whether it is supported by GNU ld.